### PR TITLE
Dev-3570 NAICS API Implementation

### PR DESCRIPTION
**Pull request by jonhill13**
**High level description:**

Adds the API call to fetch real NAICS data from the DB.

**Technical details:**

• adds naics helper

* updates fetchNAICS in NAICSContainer

* adds tests to NAICSContainer

**JIRA Ticket:**
[DEV-3570](https://federal-spending-transparency.atlassian.net/browse/DEV-3570)

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] [API contract](https://github.com/fedspendingtransparency/usaspending-api/pull/2064) updated
- [x] [API #2064](https://github.com/fedspendingtransparency/usaspending-api/pull/2064) merged
- [x] Link to this PR in JIRA ticket
- [x] [Front End PR Dev-3569](https://github.com/fedspendingtransparency/usaspending-website/pull/1294) merged

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -18,7 +19,7 @@
             }
         }
     },
-    "plugins": ["react-hooks"], // contains linting for react-hooks best practices
+    "plugins": ["react-hooks", "babel"], // contains linting for react-hooks best practices
```
Comment (MaxwellKendall) [inline .eslintrc]: wut!?! nice!!

File: src/js/components/sharedComponents/CheckboxTree.jsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        this.createNodes();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!isEqual(prevProps.nodes, this.props.nodes)) {
+            this.createNodes();
+        }
+    }
+
+    onExpand = (expanded) => this.setState({ expanded });
```
Comment (MaxwellKendall) [inline src/js/components/sharedComponents/CheckboxTree.jsx]: sweet.

File: src/js/containers/search/filters/naics/NAICSContainer.jsx [hunk 3]
```diff
@@ -397,12 +89,41 @@ export class NAICSContainer extends React.Component {
...
+            </div>
+        );
+    }
+
+    errorDiv() {
+        if (!this.state.isError) return null;
+        return (
+            <div className="container-is-loading">
+                <div className="container-is-loading__text">
+                    this.state.errorMessage
```
Comment (MaxwellKendall) [inline src/js/containers/search/filters/naics/NAICSContainer.jsx]: Brackets needed here

File: src/js/components/sharedComponents/CheckboxTreeLabel.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        PropTypes.array
+    ]),
+    label: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.array
+    ])
+};
+
+const CheckboxTreeLabel = ({
+    value,
```
Comment (MaxwellKendall) [inline src/js/components/sharedComponents/CheckboxTreeLabel.jsx]: Can this HTML be more semantic? IE <label>?

Also thinking that we should try and make this as re-usable as possible.

File: src/js/containers/shared/checkboxTree/CheckboxTree.jsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    icons: PropTypes.object,
+    nodeKeys: PropTypes.object,
+    isSearch: PropTypes.bool,
+    searchText: PropTypes.string,
+    modifyLabelTextClassname: PropTypes.string,
+    labelComponent: PropTypes.element
+};
+// default icons
+const treeIcons = {
+    check: (<FontAwesomeIcon
```
Comment (MaxwellKendall) [inline src/js/containers/shared/checkboxTree/CheckboxTree.jsx]: Should this be a dataMapping file?

File: tests/helpers/checkboxTreeHelper-test.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    naicsMockCleanData,
+    naicsMockCleanDataInitialLoad
+} from '../containers/search/filters/naics/mockNAICS';
+
+describe('CheckboxTreeHelper', () => {
+    const keysToBeMapped = {
+        value: 'naics',
+        label: 'naics_description'
+    };
+    it('should map API response to react-checkbox-tree structure', () => {
```
Comment (MaxwellKendall) [inline tests/helpers/checkboxTreeHelper-test.js]: Shouldn't we have a nested describe here.

describe(fileName)... describe(methodName)

File: tests/helpers/checkboxTreeHelper-test.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        const data = createCheckboxTreeDataStrucure(keysToBeMapped, naicsMock2);
+        expect(data).toEqual(naicsMockCleanData);
+    });
+    // for initial load of data
+    it('should map API response to react-checkbox-tree structure with empty children', () => {
+        console.log(' NAICS SHIT : ', naicsMock2);
```
Comment (MaxwellKendall) [inline tests/helpers/checkboxTreeHelper-test.js]: YOOOOOOOO

Comment (jonhill13) [reply tests/helpers/checkboxTreeHelper-test.js]: lol

File: tests/helpers/checkboxTreeHelper-test.js (hunk 6 above)
Comment (jonhill13) [reply tests/helpers/checkboxTreeHelper-test.js]: i mean, there is only one function, and two test cases. so no. maybe if we had other functions in here.

File: src/js/components/sharedComponents/CheckboxTreeLabel.jsx (hunk 4 above)
Comment (jonhill13) [reply src/js/components/sharedComponents/CheckboxTreeLabel.jsx]: Porque?

Comment (jonhill13) [reply src/js/components/sharedComponents/CheckboxTreeLabel.jsx]: no because I cannot access the input associated with that label to pass it an id

**Pull request merged**